photon-core/src/main/java/org/photonvision/vision/processes/PipelineManager.java

```diff
@@ -92,6 +92,28 @@ public CVPipelineSettings getPipelineSettings(int index) {
         return null;
     }
 
+    /**
+     * Get the settings for a pipeline by index.
+     *
+     * @param index Index of pipeline whose nickname needs getting.
+     * @return the nickname of the pipeline whose index was provided.
+     */
+    public String getPipelineNickname(int index) {
+        if (index < 0) {
+            switch (index) {
+                case DRIVERMODE_INDEX:
+                    return driverModePipeline.getSettings().pipelineNickname;
+                case CAL_3D_INDEX:
+                    return calibration3dPipeline.getSettings().pipelineNickname;
+            }
+        }
+
+        for (var setting : userPipelineSettings) {
+            if (setting.pipelineIndex == index) return setting.pipelineNickname;
+        }
+        return null;
```

Comment:
Nit: might be better to return "", as the things that call this function don't check for null